Include the CoreFoundation header because on old systems IOPowerSources.h
neglects to do so.

https://github.com/giampaolo/psutil/issues/2362
--- psutil/arch/osx/sensors.c.orig	2023-06-01 11:41:59.000000000 -0500
+++ psutil/arch/osx/sensors.c	2024-01-29 01:26:30.000000000 -0600
@@ -11,6 +11,7 @@
 // https://github.com/giampaolo/psutil/commit/e0df5da
 
 
+#include <CoreFoundation/CoreFoundation.h>
 #include <Python.h>
 #include <IOKit/ps/IOPowerSources.h>
 #include <IOKit/ps/IOPSKeys.h>
